stdlib/public/core/UnsafeRawBufferPointer.swift.gyb

```diff
@@ -337,9 +337,16 @@ extension Unsafe${Mutable}RawBufferPointer {
       _ = $0.initializeMemory(as: UInt8.self, repeating: element)
     }
 
+    var word7f = UInt(0)
+    withUnsafeMutableBytes(of: &word7f) {
+      _ = $0.initializeMemory(as: UInt8.self, repeating: 0x7f)
+    }
+
     while (i &+ stride) <= endIndex {
       let word = ptr.load(fromByteOffset: i, as: UInt.self)
-      guard word ^ mask == 0 else { break }
+      let input = word ^ mask
+      let tmp = (input & word7f) + word7f
+      guard ~(tmp | input | word7f) == 0 else { break }
```

Comment:
It seems this was difficult to express as a guard, and now the condition is negated. It would be completely fair to express this as an `if` statement, instead of having a double-negative.

Reply:
I am fine with this guard as it is.